docs/tutorial.md

```diff
@@ -0,0 +1,557 @@
+
+- [Android](https://developer.android.com/jetpack/compose/setup)
+    - Also set up [Parcelize](https://developer.android.com/kotlin/parcelize)
+- [Multiplatform](https://www.jetbrains.com/help/kotlin-multiplatform-dev/compose-multiplatform-getting-started.html)
+
+Next, add the `circuit-foundation` dependency. This includes all the core Circuit artifacts.
+
+```kotlin title="build.gradle.kts"
+dependencies {
+  implementation("com.slack.circuit:circuit-foundation:<version>")
+}
+```
+
+See [setup docs](setup.md) for more information.
+
+## Create a `Screen`
+
+The primary entry points in Circuit are `Screen`s ([docs](https://slackhq.github.io/circuit/api/0.x/circuit-runtime-screen/com.slack.circuit.runtime.screen/-screen/index.html)). These are the navigational building blocks of your app. A `Screen` is a simple data class or data object that represents a unique location in your app. For example, a `Screen` could represent an inbox list, an email detail, or a settings screen.
+
+Let's start with a simple `Screen` that represents an inbox list:
+
+=== "Android"
+    ```kotlin
+    @Parcelize
+    data object InboxScreen : Screen
+    ```
+
+=== "Multiplatform"
+    ```kotlin
+    data object InboxScreen : Screen
+    ```
+
+!!! tip
+    `Screen` is `Parcelable` on Android. You should use the Parcelize plugin to annotate your screens with `@Parcelize`.
+
+## Design your state
+
+Next, let's define some state for our `InboxScreen`. Circuit uses unidirectional data flow (UDF) to ensure strong separation between presentation logic and UI. States should be [_stable_ or _immutable_](https://developer.android.com/jetpack/compose/performance/stability), and directly renderable by your UIs. As such, you should design them to be as simple as possible.
+
+Conventionally, this is written as a nested `State` class inside your `Screen` and _must_ extend `CircuitUiState` ([docs](https://slackhq.github.io/circuit/api/0.x/circuit-runtime/com.slack.circuit.runtime/-circuit-ui-state/index.html)).
+
+=== "InboxScreen"
+    ```kotlin title="InboxScreen.kt" hl_lines="2-4"
+    data object InboxScreen : Screen {
+      data class State(
+        val emails: List<Email>
+      ) : CircuitUiState
+    }
+    ```
+
+=== "Email"
+    ```kotlin title="Email.kt"
+    @Immutable
+    data class Email(
+      val id: String,
+      val subject: String,
+      val body: String,
+      val sender: String,
+      val timestamp: String,
+      val recipients: List<String>,
+    )
+    ```
+
+See the [states and events](states-and-events.md) guide for more information.
+
+## Create your UI
+
+Next, let's define a `Ui` for our `InboxScreen`. A `Ui` is a simple composable function that takes `State` and `Modifier` parameters. It's responsible for rendering the state. You should write this like a standard composable. In this case, we'll use a `LazyColumn` to render a list of emails.
+
+// TODO side by side screenshot?
```

Comment:
We could add a preview and use a screenshot from that, good case to demo using previews for this kind of development.

Reply:
yeah I've got a couple screenshots I think I'm gonna stick in the beginning of the relevant sections, and maybe a gif, and start with that for now